Failure to copy files during CI build

## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=752591
Build error leg or test failing: Build / windows-x86 Release Libraries_NET48 / Build product
Pull request: https://github.com/dotnet/runtime/pull/105350
<!-- Error message template  -->
## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssueJsonStepByStep.md).

<!-- Use ErrorMessage for String.Contains matches. Use ErrorPattern for regex matches (single line/no backtracking). Set BuildRetry to `true` to retry builds with this error. Set ExcludeConsoleLog to `true` to skip helix logs analysis. -->

```json
{
  "ErrorMessage": "error MSB3026: Could not copy", 
  "ErrorPattern": "",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```

```
dotnet\sdk\9.0.100-preview.5.24307.3\Microsoft.Common.CurrentVersion.targets(4958,5): error MSB3026: Could not copy "D:\a\_work\1\s\artifacts\obj\System.Text.RegularExpressions.Generator\Release\netstandard2.0\fr\System.Text.RegularExpressions.Generator.resources.dll" to "D:\a\_work\1\s\artifacts\bin\System.Text.RegularExpressions.Generator\Release\netstandard2.0\fr\System.Text.RegularExpressions.Generator.resources.dll". Beginning retry 1 in 1000ms. The process cannot access the file 'D:\a\_work\1\s\artifacts\bin\System.Text.RegularExpressions.Generator\Release\netstandard2.0\fr\System.Text.RegularExpressions.Generator.resources.dll' because it is being used by another process.
```

<!-- Known issue validation start -->
 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=752591
**Error message validated:** `[error MSB3026: Could not copy`]
**Result validation:** :white_check_mark: Known issue matched with the provided build.
**Validation performed at:** 7/23/2024 11:25:56 PM UTC
<!-- Known issue validation end -->
<!--Known issue error report start -->

### Report
|Build|Definition|Step Name|Console log|Pull Request|
|---|---|---|---|---|
|[752709](https://dev.azure.com/dnceng-public/public/_build/results?buildId=752709)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/752709/logs/242)|dotnet/runtime#105357|
|[752591](https://dev.azure.com/dnceng-public/public/_build/results?buildId=752591)|dotnet/runtime|Build product|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/752591/logs/551)|dotnet/runtime#105350|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|2|2|2|
<!--Known issue error report end -->

--- a/src/libraries/oob-gen.proj
+++ b/src/libraries/oob-gen.proj
@@ -1,5 +1,11 @@
 <Project Sdk="Microsoft.Build.Traversal">
 
+  <PropertyGroup Condition="'$(BuildTargetFramework)' != '' and '$(BuildTargetFramework)' == '$(NetFrameworkCurrent)'">
+    <TargetFramework>$(BuildTargetFramework)</TargetFramework>
+    <!-- Filter ProjectReferences to build the best matching target framework only. -->
+    <FilterTraversalProjectReferences>true</FilterTraversalProjectReferences>
+  </PropertyGroup>
+
   <!-- Reference all out-of-band generator projects. -->
   <ItemGroup>
     <ProjectReference Include="$(MSBuildThisFileDirectory)*\gen\**\*.*proj"

